### PR TITLE
RSDK-7461 - Update RDK to use new API

### PR DESCRIPTION
**Pull request by nfranczak**
We update the RDK to reflect the following changes made in the API.
1. `message GeoObstacle {` is replaced by `message GeoGeometry {`
2. `MoveOnGlobeRequest` now has `common.v1.GeoGeometry bounding_regions` as an argument.

NOTE: tests will not pass until the API changes are merged here: https://github.com/viamrobotics/api/pull/495

Edit: spoke with Nick Sanford offline about how the flow of things should proceed:
1. TODO: I need to determine if the RC card will continue to work once this change is merged
2. If it does not work, then we will need  to make RC card changes and release the corresponding rdk RC card to app when RDK version which has the breaking change is promoted to stable.

### CODE REVIEW

File: spatialmath/geo_obstacle.go
Comment (biotinker) [inline spatialmath/geo_obstacle.go:1]: This file name (and the test file) should probably change names

**Review by biotinker — APPROVED** [review]
LGTM

File: services/sensors/server_test.go [hunk 1]
```diff
@@ -33,6 +32,7 @@ func TestServerGetSensors(t *testing.T) {
 		sMap := map[resource.Name]sensors.Service{}
 		server, err := newServer(sMap)
 		test.That(t, err, test.ShouldBeNil)
+		//nolint:staticcheck
```
Comment (nfranczak) [inline services/sensors/server_test.go:35]: this seems to be necessary as `make build-go lint-go` removes the top-most `//nolint:staticcheck` and then complains.
Screenshot 2024-05-14 at 1 15 32 PM

**Review by nicksanford — APPROVED** [review]
Good to go mod the bug in pbhelpers & getting tests to pass

**Pull request merged**